Add roles early access form to docs (#858)
osohq/oso: docs/content/any/guides/roles/_index.md

To learn more about roles and the different ways they can be used for authorization, head over to [Role-Based Access
Control](learn/roles).

{{% callout "Implementing roles?" "yellow" %}}

> We're working on a set of new features to better support adding roles to B2B applications.

Reviewer: ```suggestion
We're working on the next major release of Oso, a new set of features to better support adding roles to B2B applications.
```

Reviewer 2: The phrase "major release" has a specific meaning in the semver world — I could see people misintepreting this as "we're working on Oso 1.0, a new set..."

Author: Is "major update to Oso" better? Or same problem?

Reviewer 2: How about this?

```suggestion
We're working on the next major feature of Oso, a new set of tools to better support adding roles to B2B applications.
```

These features include defining roles, adding role checks/enforcement to your application, and exposing a
role management API and frontend to your end users.

> If you're interested, we encourage you to sign up for early access [here](https://osohq.typeform.com/to/w8xgMHbw)!

Reviewer: ```suggestion
Sign up for early access [here](https://osohq.typeform.com/to/w8xgMHbw)!
```
